db/wpdb.ts

```diff
@@ -660,6 +663,145 @@ export const getRelatedChartsForVariable = async (
             `)
 }
 
+interface RelatedResearchQueryResult {
+    linkTargetSlug: string
+    componentType: string
+    chartSlug: string
+    title: string
+    postSlug: string
+    chartId: number
+    authors: string
+    thumbnail: string
+    pageviews: number
+    post_source: string
+    tags: string
+}
+export const getRelatedResearchAndWritingForVariable = async (
+    variableId: number
+): Promise<DataPageRelatedResearch[]> => {
+    const wp_posts: RelatedResearchQueryResult[] = await db.queryMysql(
+        `-- sql
+            select
+                distinct
+                pl.target as linkTargetSlug,
+                pl.componentType as componentType,
+                c.slug as chartSlug,
+                p.title as title,
+                p.slug as postSlug,
+                coalesce(csr.chart_id, c.id) as chartId,
+                p.authors as authors,
+                p.featured_image as thumbnail,
+                coalesce(pv.views_365d, 0) as pageviews,
+                'wordpress' as post_source,
+                (select coalesce(JSON_ARRAYAGG(t.name), JSON_ARRAY())
+                    from post_tags pt
+                    join tags t on pt.tag_id = t.id
+                    where pt.post_id = p.id
+                ) as tags
+            from
+                posts_links pl
+            join posts p on
+                pl.sourceId = p.id
+            left join charts c on
+                pl.target = c.slug
```

Comment:
If I see this right this will only work if there's _not_ a redirect?
And otherwise, `chartSlug` above will be null, and the join of `chart_dimensions` will also not work?

Reply:
Great catch, thanks!